net-mgmt/pfSense-pkg-ANDwatch/files/usr/local/pkg/andwatch.inc

```diff
@@ -0,0 +1,196 @@
+<?php
+/*
+ * andwatch.inc
+ *
+ * part of pfSense (https://www.pfsense.org)
+ * Copyright (c) 2025 Denny Page
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+require_once("config.inc");
+require_once("functions.inc");
+require_once("util.inc");
+require_once("service-utils.inc");
+
+const ANDWATCH_SVC_NAME = 'andwatchd';
+const ANDWATCH_RC_FILE = '/usr/local/etc/rc.d/andwatch.sh';
+const ANDWATCHD_CMD = '/usr/local/bin/andwatchd';
+const ANDWATCHD_NOTIFY_CMD = '/usr/local/pkg/andwatch-notify.php';
+const ANDWATCHD_PID_PREFIX = '/var/run/andwatch-';
+const ANDWATCH_MA_DB_FILE = '/var/db/andwatch/ma_db.sqlite';
+const ANDWATCH_UPDATE_MA_CMD = '/usr/local/bin/andwatch-update-ma';
+const ANDWATCH_QUERY_CMD = '/usr/local/bin/andwatch-query';
+
+$shortcut_section = 'andwatch';
+
+// Is ANDwatch enabled?
+function andwatch_enabled() {
+	return (config_get_path('installedpackages/andwatch/enable', false));
+}
+
+
+// Write the rc file
+function andwatch_write_rcfile() {
+	// Get the current configuration
+	$current_config = config_get_path('installedpackages/andwatch');
+
+	// Get the real interface names
+	$interfaces = array();
+	foreach (explode(',', array_get_path($current_config, 'active_interfaces', '')) as $interface) {
```

Comment:
The function `explode()` used with an empty string will result in an array with a single empty string element. Use `array_filter(explode(<...>))` to avoid this.